app/code/Magento/Paypal/Model/Api/Nvp.php

```diff
@@ -1488,7 +1488,24 @@ protected function _exportAddresses($data)
             \Magento\Framework\DataObject\Mapper::accumulateByMap($data, $shippingAddress, $this->_shippingAddressMap);
             $this->_applyStreetAndRegionWorkarounds($shippingAddress);
             // PayPal doesn't provide detailed shipping name fields, so the name will be overwritten
-            $shippingAddress->addData(['firstname'  => $data['SHIPTONAME']]);
+            $_arraydata = explode(' ', $data['SHIPTONAME']);
```

Comment:
```suggestion
            $nameParts = explode(' ', $data['SHIPTONAME']);
```
Please use a more descriptive variable name.